### PR TITLE
ASoC: SOF: tgl: set use_acpi_target_states for the tgl platforms

### PR DESCRIPTION
**Pull request by libinyang**
All Intel tgl platforms should support the feature of getting the
system state from acpi.

Signed-off-by: Libin Yang <libin.yang@intel.com>

### CODE REVIEW

**Comment by libinyang** [comment]
use_acpi_target_states is used in sof power management. When snd_sof_prepare() is called, the state may be SOF_SUSPEND_S3 or SOF_SUSPEND_S0IX. The use_acpi_target_states is used to determine whether to check the state is SOF_SUSPEND_S0IX or not.

**Comment by bardliao** [comment]
@libinyang How do we know which platform needs the use_acpi_target_states flag?

**Review by kv2019i — DISMISSED** [review]

File: sound/soc/sof/intel/pci-tgl.c [hunk 1]
```diff
@@ -39,6 +39,7 @@ static const struct sof_dev_desc tgl_desc = {
 static const struct sof_dev_desc tglh_desc = {
 	.machines               = snd_soc_acpi_intel_tgl_machines,
 	.alt_machines		= snd_soc_acpi_intel_tgl_sdw_machines,
+	.use_acpi_target_states	= true,
```
Comment (plbossart) [inline sound/soc/sof/intel/pci-tgl.c:42]: that's a valid fix, but it needs to appear as such to maintainers.

Please reword the commit message, e.g.

ASoC: SOF: tgl: add missing use_acpi_target_states for the tgl platforms

All Intel tgl platforms should support the feature of getting the system state from acpi to deal with S0ix support.
This was missed in previous commits, likely due to copy/paste from older code.

**Review by plbossart — COMMENTED** [review]
this is a good fix but we need to be clearer on why this is needed.

**Comment by libinyang** [comment]
> @libinyang How do we know which platform needs the use_acpi_target_states flag?

@bardliao I think most of our platforms which support ACPI and S0ix should set this flag.

File: sound/soc/sof/intel/pci-tgl.c (hunk 1 above)
Comment (libinyang) [reply sound/soc/sof/intel/pci-tgl.c:42]: @plbossart Thanks for the suggestion of the commit messages.

**Review by plbossart — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Pull request merged**